### DISCUSSION TITLE
Progress Bar

### DISCUSSION THREAD

**Discussion by KemenSarn** [missionpinball/missionpinball#19] [General]
Hi All,

I posted this in mpf users but i'll do it here as well.

I have read a few threads in here and can get this to work with hard coded variables, but when I try and make it dynamic it crashes mc

So I have a show.yaml with

#show_version=5

- duration: -1
  slides:
    mode_base_started:
      widgets:
      - type: rectangle
        color: red
        anchor_x: 0
        anchor_y: 0
        x: 50%
        y: 50%
        z: 500
        width: (bar_width)
        height: 30
        animations:
          update_boss_width:
            - property: width
              value: 50

and in my config I have:

show_player:
  mode_boss_started:
    boss_progress:
      show_tokens:
        bar_width: 300
...

event_player:
  shot_hit:
    - update_boss_width

so this works and sets the bar width to 50

when i try and add a dynamic value into the width property from my config is when it is crashing.

show yaml

        animations:
          update_boss_width:
            - property: width
              value: (new_bar_width)

config

event_player:
 shot_hit:
    - boss_hit
    - update_boss_width

variable_player:
  boss_hit:
    new_bar_width: 50

this crashes

any help would be great.

Kemen

**Comment by Pucksnok** [comment]
What does the crash tell? Run with -t or check logs

**Comment by KemenSarn** [comment]
So the log is telling me its is updating the variable
INFO : EventManager : Event: ======'player_new_bar_width'====== Args={'value': 400, 'prev_value': 300, 'change': 100, 'player_num': 1}
EventManager: Event: ======'player_new_bar_width'====== Args={'value': 400, 'prev_value': 300, 'change': 100, 'player_num': 1}

but when the update_bar_width event triggers I get:

EventManager: Event: ======'update_boss_width'====== Args={'priority': 0}
kivy: Base: Leaving application in progress...
kivy: Exception while processing RegisteredHandler(callback=<bound method Widget.start_animation_from_event of <Rectangle Widget id=2158210936>>, priority=1, kwargs={'event_name': 'update_boss_width'}, key=UUID('f9da3093-1d98-4075-b1f6-a71a9fbd5e6b'), condition=None, blocking_facility=None) for event update_boss_width. Excepted an event parameter player_new_bar_width
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/mpfmc/uix/widget.py", line 457, in _convert_animation_value_to_float
    val = event_args[val[1:-1]]
KeyError: 'player_new_bar_width'

  File "/usr/local/lib/python3.9/dist-packages/mpf/core/events.py", line 697, in _run_handlers
    raise Exception("Exception while processing {} for event {}. {}".format(handler, event, e)) from e
Exception: Exception while processing RegisteredHandler(callback=<bound method Widget.start_animation_from_event of <Rectangle Widget id=2158210936>>, priority=1, kwargs={'event_name': 'update_boss_width'}, key=UUID('f9da3093-1d98-4075-b1f6-a71a9fbd5e6b'), condition=None, blocking_facility=None) for event update_boss_width. Excepted an event parameter player_new_bar_width



which seems like a similar error from the other threads that was fixed in a release.

**Comment by KemenSarn** [comment]
Ok so I took what I said and passed the value in the event player, and surprise surprise it works...

Next issue...how to return the current width parameter of the widget???
